### PR TITLE
Update XSL options and add admin form select

### PR DESCRIPTION
**Pull request by bondjimbond**
**JIRA Ticket**: (https://jira.duraspace.org/browse/ISLANDORA-2404)

# What does this Pull Request do?

Adds the latest versions of the LoC MARC-to-MODS and MODS-to-MARC XSL files, with admin form select options so you can still use the old ones if you prefer them.

# What's new?

- New XSL files from official LOC source
- New admin form options so you can select them
- Default value updated to the latest version in XSLT 1.0 to minimize any potential (unlikely) conflicts

# How should this be tested?

- Get a sample MARCXML file (eg http://ebba.english.ucsb.edu/content/marcxml.html)
- Configure islandora_marcxml to use different versions of the XSLT each time you ingest an object using this file
- Confirm you've got valid data
- View an object's MARCXML tab and make sure that you are still getting valid MARC after changing MODS-to-MARC versions.

# Interested parties
@Islandora/7-x-1-x-committers

### CODE REVIEW

**Comment by patdunlavey** [comment]
I'll review this today since we need it!

**Comment by patdunlavey** [comment]
Apologies, I misunderstood what this PR does. I thought it provided the ability for users to substitute their own customized xsl files. I do agree with the intent of this PR, but I have not tested it.

**Comment by jordandukart** [comment]
Will try and get to this today @bondjimbond

File: includes/admin.form.inc
Comment (jordandukart) [inline includes/admin.form.inc:21]: So the "old" included files are no longer choices? Typically I'd think they'd be the default choice.

File: includes/admin.form.inc
Comment (jordandukart) [inline includes/admin.form.inc]: PHP's native XSLTProcessor can't transform XSL 2.0 stylesheets. An external tool like `saxonb-xslt` would need to be used to handle them which is something that core Islandora currently does not support. Should potentially remove them?

**Review by jordandukart — CHANGES_REQUESTED** [review]

File: includes/admin.form.inc
Comment (bondjimbond) [reply includes/admin.form.inc:21]: The old files are still there, but the newest edition of the same schema is the new default.

File: includes/admin.form.inc
Comment (bondjimbond) [reply includes/admin.form.inc]: Sounds good! Addressed with latest commit.

File: includes/admin.form.inc
Comment (patdunlavey) [inline includes/admin.form.inc]: Shouldn't this be:
`    '#default_value' => variable_get('islandora_marcxml_select_mods_to_marc_xsl', 'MODS3-4_MARC21slim_XSLT1-0.xsl'),`

File: includes/admin.form.inc
Comment (patdunlavey) [inline includes/admin.form.inc]: Shouldn't this be
`    '#default_value' => variable_get('islandora_marcxml_select_marc_to_mods_xsl', 'MARC21slim2MODS3-6.xsl'),
`

**Comment by patdunlavey** [comment]
@bondjimbond I'm going to send you a pull-request on your fork in order to get support for user-entered paths to other xslt files. It'll just take me a few to write that up and send your way. It depends on installing the select_or_other module. You'll see...
Pat

**Comment by bondjimbond** [comment]
@patdunlavey I think it's better to deal with/merge this one first, and make user-entered paths a second PR that builds on the changes from this one -- it's a bigger request and may need more discussion.

**Comment by patdunlavey** [comment]
@bondjimbond ok, I'll consider my pull request tabled. I'll comment on the JIRA ticket.

File: includes/admin.form.inc
Comment (bondjimbond) [reply includes/admin.form.inc:21]: Also, the spirit of this ticket is to update the transforms to the most current ones, which I believe should be the default behaviour. I added the option to select the outdated transforms as sort of a bonus feature because some sites don't like to change things -- I believe the majority would want the latest version by default, rather than having to change configuration to choose it.

**Comment by bondjimbond** [comment]
Note from Committers' Call: Suggestion that it would be nice to include an update hook to notify folks that the update happened.

**Comment by bondjimbond** [comment]
@jordandukart Finally got round to adding the update hook, after we talked about this two calls ago. All concerns have been addressed, I believe. Any other tweaks required?

**Comment by bondjimbond** [comment]
Just a bump - I think everything's been addressed, so it's just waiting for approval/merge.

**Comment by bondjimbond** [comment]
@jordandukart or @patdunlavey Any further thoughts on this? Still waiting on approval.

**Comment by bondjimbond** [comment]
@Islandora/7-x-1-x-committers @patdunlavey @jordandukart 
This has been in limbo for a while, and I think all issues have been resolved. Can it be approved or further critiqued?

File: includes/admin.form.inc
Comment (bondjimbond) [reply includes/admin.form.inc:21]: Note also that in past updates to Official LoC XSLs, we have just straight up updated them -- rather than providing the choice to keep using the old ones.

Everything has been addressed as far as I see. Are you able to approve?

**Comment by whikloj** [comment]
We're not a user of the MarcXML components in Islandora, but I am wondering if this default is going to change again and if so, should you assign it to a `const`ant and use that constant for any `variable_get`s?

Then if you need to move up to a newer default, you can just update the filename in two locations instead of more?

**Comment by jordandukart** [comment]
I think the intent would be that as a newer version needs to be added another option occurs in that select such that you aren't necessarily changing the XSL is in use without the end user knowing. 

@bondjimbond I'll try free some time to test this today, apologies.

**Review by jordandukart — APPROVED** [review]
Tested, fine by me. Will wait another 24 hrs before pressing the button.

**Pull request merged**
